crates/common/src/compile.rs

```diff
@@ -284,32 +284,45 @@ pub fn compile_with_filter(
     ProjectCompiler::with_filter(print_names, print_sizes, skip).compile(project)
 }
 
-/// Compiles the provided [`Project`], throws if there's any compiler error and logs whether
+/// Compiles the provided [`Project`], throws if there's any compiler error (if throw err is true) and logs whether
 /// compilation was successful or if there was a cache hit.
 /// Doesn't print anything to stdout, thus is "suppressed".
-pub fn suppress_compile(project: &Project) -> Result<ProjectCompileOutput> {
+pub fn suppress_compile(project: &Project, throw_err: bool) -> Result<ProjectCompileOutput> {
```

Comment:
not a fan of this solution,

I'd like to add this feature as an additional function

`fn try_suppress_compile` that simply doesn't check the compiler errors. `suppress_compile` can then call this and apply the error check